plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -215,7 +215,19 @@ export class DeploymentManager {
     const maybeExisting = await this.contract<C>(alias);
     if (!maybeExisting) {
       const trace = this.tracer();
-      const address = await this.readAlias(network, deployment, otherAlias);
+      let address = await this.readAlias(network, deployment, otherAlias);
+      if (address == null) {
+        const dm = new DeploymentManager(
+          network,
+          deployment,
+          this.hre,
+          {
+            writeCacheToDisk: true,
+          }
+        );
+        await dm.spider();
+        address = await this.readAlias(network, deployment, otherAlias);
+      }
```

Comment:
As is we call this multiple times, so I think we'd at least want caching with this approach, but kinda like calling it from the script before

Reply:
So you prefer spidering from the deploy script? I'm on board with that too.

This approach does cache already right? `writeCacheToDisk` is `true`, so any future spiders won't make network calls to etherscan.

Reply:
Yea we cache network requests, but still does a bunch of work / noise on each call. Perhaps not so bad if limited to a handful of calls in deploy, but also imagine it may pop up in migration scripts

Reply:
Yeah definitely could be optimized further. I got rid of this change for now and moved the spidering into the deploy script.